packages/common/editors/EditorPortal.js

```diff
@@ -2,24 +2,36 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 
-const editorRoot = document.body;
-
 export default class EditorPortal extends React.Component {
   static propTypes = {
-    children: PropTypes.node
+    children: PropTypes.node.isRequired,
+    target: PropTypes.instanceOf(Element).isRequired
+  };
+
+  // Keep track of when the modal element is added to the DOM
+  state = {
+    isMounted: false
   };
 
   el = document.createElement('div');
 
   componentDidMount() {
-    editorRoot.appendChild(this.el);
+    this.props.target.appendChild(this.el);
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({ isMounted: true });
```

Comment:
Does this need to be in the state? I wonder if we can just do `this.isMounted = true`

Reply:
We need to trigger an update when the element is attached to DOM. With instance variable react would not know that we want to rerender the component